server/routes/index.ts

```diff
@@ -35,6 +36,20 @@ export default function routes(services: Services): Router {
   router.get('/plan/create/:id/**/update', [retrieveInduction(services.ciagService, services.inductionService)])
   router.post('/plan/create/:id/**/update', [retrieveInduction(services.ciagService, services.inductionService)])
 
+  // The check-your-answers page is used for both the Create and Update journey
+  // When the page is submitted (POST) we need to have retrieved the induction, but only in the case of the Update journey
+  router.post('/plan/create/:id/check-your-answers', [
+    async (req: Request, res: Response, next: NextFunction) => {
+      const { id } = req.params
+      const isUpdateFlow = getSessionData(req, ['isUpdateFlow', id])
+      if (!isUpdateFlow) {
+        next()
+      } else {
+        retrieveInduction(services.ciagService, services.inductionService)(req, res, next)
+      }
+    },
+  ])
+
```

Comment:
This is the fix. It sucks because we cant really unit test it because its an anonymous function; hence the need for the cypress test changes and additions that I've made.

Basically what it does is bind to the `POST check-your-answers` page. It first checks whether this is the "update flow" by getting a property from the session (that thankfully Gary setup !)
If it's not the update flow (ie. it must be the create flow, then just bail out by calling the `next` function

If it is the update flow, then call the middleware to retrieve the induction

Reply:
The crux of this is actually that the update flow was shoe-horned in after-the-event ("with no requirements or design"), and that the same route path and controller method has been used to handle both the Create and Update journey.
If they both had their own route path and a different controller method, then this would be much easier - we wouldn't need this logic in the route definition and the whole thing would be much easier to unit test.

Perhaps when we look at the controller and the Update functionality within it (looking at you @mattwills-moj-digital 😁 ) , maybe we'll look at refactoring some of it out, which might go a way towards addressing this 🤞

Reply:
Perhaps this is something we can look at tomorrow 😄